evm_arithmetization/src/generation/prover_input.rs

```diff
@@ -757,7 +765,34 @@ impl<F: RichField> GenerationState<F> {
     fn generate_jumpdest_table(&mut self) -> Result<(), ProgramError> {
         // Simulate the user's code and (unnecessarily) part of the kernel code,
         // skipping the validate table call
-        self.jumpdest_table = simulate_cpu_and_get_user_jumps("terminate_common", self);
+
+        info!("Generating JUMPDEST tables");
+        // dbg!(&self.inputs.jumpdest_table);
+        // w for witness
+        // let txn_idx = self.next_txn_index - 1;
+        // let rpcw = self.inputs.jumpdest_tables[txn_idx].as_ref();contract_code
+        let rpcw = &self.inputs.jumpdest_table;
+        let rpc: Option<JumpDestTableProcessed> = rpcw
+            .as_ref()
+            .map(|jdt| set_jumpdest_analysis_inputs_rpc(jdt, &self.inputs.contract_code));
+
+        let sims = simulate_cpu_and_get_user_jumps("terminate_common", self);
+
+        let (sim, simw): (Option<JumpDestTableProcessed>, Option<JumpDestTableWitness>) =
+            sims.map_or_else(|| (None, None), |(sim, simw)| (Some(sim), Some(simw)));
+
+        if let (Some(rw), Some(sw)) = (rpcw, simw)
+            && rw != &sw
+        {
+            info!("SIMW {}", sw);
+            info!("RPCW {}", rw);
+            assert_eq!(rw, &sw);
+        }
+
+        info!("SIM {:#?}", sim);
+        info!("RPC {:#?}", rpc);
+
+        self.jumpdest_table = if rpc.is_some() { rpc } else { sim };
```

Comment:
would need to change, as there is no simulation if the table exists

Reply:
This will indeed be removed.  I just need to keep it around until we are happy with the results, so I can use the simulation output as ground truth.